nrf_modem/doc/modem_trace.rst

```diff
@@ -0,0 +1,67 @@
+.. _modem_trace:
+
+.. _%XMODEMTRACE AT command documentation: https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/mob_termination_ctrl_status/xmodemtrace.html
+
+Modem traces
+############
+
+The modem trace interface in the Modem library is used to retrieve trace data from the modem core.
+
+Enabling modem traces
+*********************
+
+The modem traces are enabled with the AT command ``%XMODEMTRACE``, with the option to select multiple output levels from the modem.
+For a complete overview, see the `%XMODEMTRACE AT command documentation`_.
+
+Handling modem traces
+*********************
+
+Modem traces are pulled by the application and output to a trace backend, typically UART.
+The traces can be retrieved from the Modem library by calling :c:func:`nrf_modem_trace_get`.
+
+The trace data, containing multiple fragments, must be processed in order, starting from the first array index.
+
+Once trace data is processed by the trace backend, the :c:func:`nrf_modem_trace_processed_callback` must be called to free up the memory allocated in the trace memory area.
+If the trace data is processed in chuncs, this can be split into multiple calls, e.g. one call for each fragment of processed data.
+
+.. figure:: images/nrf_modem_trace_sequence_data_available.svg
+   :alt: Modem library trace processing
+
+   Modem library trace processing
+
+.. note::
+
+	If no trace data is available, :c:func:`nrf_modem_os_trace_get` will not return before new data becomes available or the modem is shut down or has faulted.
+	This require the traces to be processed in its own thread separate from the rest of the application.
+
+Asynchronous trace processing
```

Comment:
I would rephrase this chapter a bit, and I am not sure about the chapter's tile.
The important information in this chapter is how to use `nrf_modem_trace_processed_callback`

I would say:

As the application processes the trace data, the application must call the `nrf_modem_trace_processed_callback` function to inform the Modem library that the trace data has been handled and can be freed. It is not necessary that the application has processed a whole "trace fragment" before calling `nrf_modem_trace_processed_callback`. To maximize the availability of the shared memory in the trace region, the application should call `nrf_modem_trace_processed_callback` as soon as some bytes are processed, as often as possible.

Reply:
I agree on the title, perhaps `Freeing processed traces` can be a better title?

I will make the other changes as well.